### DISCUSSION TITLE
multipe MVT Layers

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#5563] [Q&A] [answered]
The MVT layer is great, but I want to know if there are any tutorials or examples of how to compose multiple MVT layers (e.g. a layer for a scatterplot and a layer for polygons).

For example, I have this [example](https://observablehq.com/@cornhundred/deck-gl-mvt-layer-nr) where I am trying to set up multiple scatterplots based on multiple MVT layers. 

```
{ 
  
  // Non-Residential Layer
  const nr_layer = new deck.MVTLayer({
    id: 'MVTLayer',
    data: [
      `https://raw.githubusercontent.com/cornhundred/vector_tile_pbf_test/master/NR2019-pC-PBF/{z}/{x}/{y}.pbf`
    ],
    getLineColor: [0, 0, 255],
    getFillColor:[0, 0, 255],
    lineWidthMinPixels: 1,
    // lineWidthPixels:0,
    // radiusMinPixels: 10,
    getRadius:5,
    pickable: true,    
  });  
  
  // Residential Layer
  const re_layer = new deck.MVTLayer({
    id: 'MVTLayer',
    data: [      
      `https://raw.githubusercontent.com/cornhundred/vector_tile_pbf_test/master/RE2019-pC-PBF/{z}/{x}/{y}.pbf` 
    ],
    getLineColor: [255, 0, 0],
    getFillColor:[255,0, 0],
    lineWidthMinPixels: 1,
    // lineWidthPixels:0,
    // radiusMinPixels: 10,
    getRadius:5,
    pickable: true,    
  });  
  
  var layers = [nr_layer, re_layer]
  
  deckgl.setProps({layers: layers});
}
```

But the first layer seems to be the only one showing up. Should I be using `renderSubLayers` or something?

**Comment by Pessimistress** [comment, marked as answer]
Layer ids must be unique.

I suggest you start by reading deck.gl's [developer guide](https://deck.gl/docs/developer-guide/using-layers).

**Reply by cornhundred** [reply]
Thanks for the help and advice @Pessimistress!!
